libraries/entities/src/index.js

```diff
@@ -23,7 +26,54 @@ class Entity {
 
   create(properties, overwrite) {
     const documentId = this.calculateId(properties);
-    this.db.procs[this.serviceName][`${this.tableName}_create`](documentId, properties, overwrite, 1);
+
+    return this.db.procs[`${this.tableName}_create`](documentId, properties, overwrite, 1);
+  }
+
+  delete(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_delete`](documentId);
+  }
+
+  update(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_update`](documentId, properties, 1);
+  }
+
+  load(properties) {
+    const documentId = this.calculateId(properties);
+
+    return this.db.procs[`${this.tableName}_load`](documentId);
+  }
+
+  async scan(options = {}) {
+    const {
+      filter,
+      limit = 1000,
+      page = 1,
+    } = options;
+
+    if (filter) {
+      assert(typeof filter === 'function', 'filter should be a function');
+    }
+
+    const result = await this.db._withClient(READ, client => {
```

Comment:
We'll need to avoid using raw clients for this library, and stick to stored procedures.  Otherwise, the code using the raw clients will stop working when we migrate away from the entities tables, as the tables will no longer exist. So, scanning and querying will need to be done in a stored procedure.

Reply:
You make a good point. If anything in the db changes, queries made with the raw client may break.

> the raw clients will stop working when we migrate away from the entities tables, as the tables will no longer exist.

In that case, would it make sense to have our stored procedures not have as a prefix the table name? Right now, `create`, `delete`, `update`, and `load` all do ``` return this.db.procs[`${this.tableName}_xxxx`](...); ```

Reply:
Yeah, maybe, if they can take the table name as a parameter.  I can imagine a few different approaches.  We can talk about that this afternoon.